src/components/v2/AuthModal/index.tsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+
+import { Modal, IModalProps } from '../Modal';
+import { AccountDetails, IAccountDetailsProps } from './AccountDetails';
+import { WalletList, IWalletListProps } from './WalletList';
+
+export interface IAuthModalProps {
+  isOpen: boolean;
+  onClose: IModalProps['handleClose'];
+  onLogin: IWalletListProps['onLogin'];
+  onLogOut: IAccountDetailsProps['onLogOut'];
+  onCopyAccountAddress: IAccountDetailsProps['onCopyAccountAddress'];
+  account?: IAccountDetailsProps['account'];
+}
+
+export const AuthModal: React.FC<IAuthModalProps> = ({
```

Comment:
Since this is being rendered out side the hierarchy of the route we could even use a portal

Reply:
Using a portal there is a valid point, but wouldn't we want the `Modal` component to be responsible of that?

Reply:
yes, I think that makes sense